DICOM/src/Microsoft.Health.DeID.SharedLib/CryptoHashFunction.cs

```diff
@@ -8,67 +8,71 @@
 using System.Security.Cryptography;
 using System.Text;
 using EnsureThat;
+using Microsoft.Health.DeID.SharedLib.Settings;
 
 namespace Microsoft.Health.Dicom.DeID.SharedLib
 {
     public class CryptoHashFunction
     {
-        public static byte[] ComputeHmacSHA256Hash(byte[] input, byte[] hashKey = null)
+        private HMAC _hmac;
+
+        public CryptoHashFunction(CryptoHashSetting cryptoHashSetting = null)
         {
-            if (input == null)
+            cryptoHashSetting ??= new CryptoHashSetting();
+            if (cryptoHashSetting.CryptoHashKey != null)
             {
-                return input;
+                _hmac = new HMACSHA256(Encoding.UTF8.GetBytes(cryptoHashSetting.CryptoHashKey));
             }
-
-            HMAC hmac = new HMACSHA256();
-            if (hashKey != null)
+            else
             {
-                hmac = new HMACSHA256(hashKey);
+                _hmac = new HMACSHA256();
```

Comment:
We can use _hmac = condition ? a : b to make this simpler.

Reply:
Thanks, fixed it